src/ImageSharp/Color/Color.cs

```diff
@@ -240,28 +261,38 @@ public TPixel ToPixel<TPixel>()
             Span<TPixel> destination)
             where TPixel : unmanaged, IPixel<TPixel>
         {
-            ReadOnlySpan<Rgba64> rgba64Span = MemoryMarshal.Cast<Color, Rgba64>(source);
-            PixelOperations<TPixel>.Instance.FromRgba64(configuration, rgba64Span, destination);
+            Guard.DestinationShouldNotBeTooShort(source, destination, nameof(destination));
+            for (int i = 0; i < source.Length; i++)
+            {
+                destination[i] = source[i].ToPixel<TPixel>();
+            }
```

Comment:
Just a note that it's possible to detect the common `boxedHighPrecisionPixel  == null` case with SIMD and bulk convert, if we really want to.

Reply:
Wasn’t aware of that. SIMD is still very much a Google event for me

Reply:
`object` is basically a pointer so it's unsafe convertible to `long` on 64 bits. We can check 4 `Color`s with bitwise masks, if all of them are NULL we can shuffle stuff so the `data` fields are together, then bulk convert them.

Super unsure if it's worth the trouble.

Reply:
>Super unsure if it's worth the trouble.

Yeah, not attempting that just now. Not worth the trouble.